File: gcj_2018/QR_B.cpp
// Google Code Jam 2018 Qualification Round
// Problem B. Trouble Sort

#include <iostream>
#include <sstream>
#include <vector>
#include <algorithm>

using namespace std;

int solve(vector<int> &v) {
	vector<int> a[2];
	for (int i = 0; i != v.size(); ++i) {
		a[i % 2].push_back(v[i]);
	}
	for (int i = 0; i < 2; ++i) {
		sort(a[i].begin(), a[i].end());
	}
	vector<int> sorted = v;
	sort(sorted.begin(), sorted.end());
	for (int i = 0; i != v.size(); ++i) {
		if (sorted[i] != a[i % 2][i / 2]) {
			return i;
		}
	}
	return -1;
}

int main(int argc, char *argv[]) {
	int T;
	cin >> T;
	for (int t = 1; t <= T; ++t) {
		int n;
		cin >> n;
		vector<int> v(n);
		for (int i = 0; i < n; ++i) {
			cin >> v[i];
		}
		int ans = solve(v);
		if (ans < 0) {
			cout << "Case #" << t << ": OK" << endl;
		} else {
			cout << "Case #" << t << ": " << ans << endl;
		}
	}
	return 0;
}

File: gcj_2018/QR_B.go
// Google Code Jam 2018 Qualification Round
// Problem B. Trouble Sort

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"sort"
)

var sc = bufio.NewScanner(os.Stdin)

func nextString() string {
	sc.Scan()
	return sc.Text()
}

func nextInt() int {
	i, e := strconv.Atoi(nextString())
	if e != nil {
		panic(e)
	}
	return i
}

func solve(n int, v []int) int {
	a := make([][]int, 2)
	for i := 0; i < n; i++ {
		a[i % 2] = append(a[i % 2], v[i])
	}
	sort.Sort(sort.IntSlice(v))
	sort.Sort(sort.IntSlice(a[0]))
	sort.Sort(sort.IntSlice(a[1]))

	for i := 0; i < n; i++ {
		if v[i] != a[i % 2][i / 2] {
			return i
		}
	}

	return -1
}

func main() {
	sc.Split(bufio.ScanWords)
	T := nextInt()
	for t := 1; t <= T; t++ {
		n := nextInt()
		v := make([]int, n)
		for i := 0; i < n; i++ {
			v[i] = nextInt()
		}

		ans := solve(n, v)

		fmt.Print("Case #")
		fmt.Print(t)
		fmt.Print(": ")
		if ans < 0 {
			fmt.Print("OK")
		} else {
			fmt.Print(ans)
		}
		fmt.Println("")
	}
}
